monasca_events_api/tests/test_events.py

```diff
@@ -0,0 +1,330 @@
+        res.body = {}
+        res.status = 0
+        try:
+            eventsObj.on_post(self._generate_req(), res)
+            assertFalse(
+                1,
+                msg="Get Method should fail but succeeded, bad request sent")
+        except Exception as e:
+            self.assertRaises(falcon.HTTPBadRequest)
+            self.assertEqual(e.status, '400 Bad Request')
+
+    @mock.patch(
+        'monasca_events_api.common.messaging.kafka_publisher.KafkaPublisher')
+    @mock.patch('monasca_events_api.v2.common.helpers.validate_authorization')
+    @mock.patch('monasca_events_api.v2.events.Events._validate_event')
+    @mock.patch('monasca_events_api.v2.common.helpers.read_http_resource')
+    @mock.patch(
+        'monasca_events_api.v2.common.helpers.validate_json_content_type')
+    @mock.patch('monasca_events_api.v2.common.helpers.get_tenant_id')
+    def test_on_post_kafka_down(
+            self,
+            tenantid,
+            json,
+            readHttpRes,
+            event,
+            validate,
+            kafka):
+        """POST method with Kafka Down"""
+        kafka.send_message.side_effect = exceptions.MessageQueueException()
+
+        json.return_value = None
+        validate.return_value = True
+        readHttpRes.return_value = self._generate_req()
+        tenantid.return_value = '0ab1ac0a-2867-402d'
+        event.return_value = True
+
+        eventsObj = EventsSubClass()
+        eventsObj._message_queue = kafka
+        res = mock.MagicMock()
+        res.body = {}
+        try:
+            eventsObj.on_post(self._generate_req(), res)
+            assertFalse(
+                1,
+                msg="Kakfa Server Down, Post should fail but succeeded")
+        except Exception as e:
+            self.assertRaises(falcon.HTTPInternalServerError)
+            self.assertEqual(e.status, '500 Internal Server Error')
+
+    @mock.patch(
+        'monasca_events_api.common.messaging.kafka_publisher.KafkaPublisher')
+    @mock.patch('monasca_events_api.v2.common.helpers.validate_authorization')
+    @mock.patch('monasca_events_api.v2.events.Events._validate_event')
+    @mock.patch('monasca_events_api.v2.common.helpers.read_http_resource')
+    @mock.patch(
+        'monasca_events_api.v2.common.helpers.validate_json_content_type')
+    @mock.patch('monasca_events_api.v2.common.helpers.get_tenant_id')
+    def test_on_post_pass(self, tenantid, json, http, event, validate, kafka):
+        """POST method success """
+        json.return_value = None
+        validate.return_value = True
+        http.return_value = True
+        tenantid.return_value = '0ab1ac0a-2867-402d'
+        event.return_value = True
+
+        eventsObj = EventsSubClass()
+        eventsObj._message_queue = kafka
+        res = mock.MagicMock()
+        res.body = {}
+        res.status = 0
+        eventsObj.on_post(self._generate_req(), res)
+        self.assertEqual(falcon.HTTP_204, res.status)
```

Comment:
This is the correction for test stream defintions (not events)

Your code has invalid data in readjson.return_value.

def test_on_post_pass(
            self,
            validate,
            tenantid,
            json,
            readjson,
            streamvalid,
            getname,
            desc,
            fireactions,
            expire,
            streamsrepo,
            kafka,
            addlink):
        """POST method successful"""
        validate.return_value = True
        fireactions.return_value = "fire_actions"
        getname.return_value = "Test"
        addlink.return_value = "/v2.0/stream-definitions/{stream_id}"
        expire.return_value = "expire_actions"
        desc.return_value = "Stream_Description"
        streamsrepo.return_value = True
        readjson.return_value = {u'name': "Test",
                                 u'id': "1",
                                 u'description': "Stream_Description",
                                 u'select': "select",
                                 u'group_by': "group_by",
                                 u'fire_criteria': "fire_criteria",
                                 u'expiration': "expiration",
                                 u'fire_actions': "fire_actions",
                                 u'expire_actions': "expire_actions",
                                 u'actions_enabled': u'true'}

From:  angelomendonca <notifications@github.com>
Reply-To:  hpcloud-mon/monasca-events-api
<reply+007359b4710f390d96402204e42b97a6ea1b8df25251a20e92cf000000011199a49b9
2a170ce01f296bb@reply.github.com>
Date:  Wednesday, June 17, 2015 at 3:11 PM
To:  hpcloud-mon/monasca-events-api <monasca-events-api@noreply.github.com>
Cc:  Cindy ONeill <cindy.o-neill@hp.com>
Subject:  Re: [monasca-events-api] Added test for GET and POST methods in
events api (#2)

> In monasca_events_api/tests/test_events.py
> <https://github.com/hpcloud-mon/monasca-events-api/pull/2#discussion_r32675515
> > :
>> > +    @mock.patch('monasca_events_api.v2.common.helpers.get_tenant_id')
>> > +    def test_on_post_pass(self, tenantid, json, http, event, validate,
>> kafka):
>> > +        """POST method success """
>> > +        json.return_value = None
>> > +        validate.return_value = True
>> > +        http.return_value = True
>> > +        tenantid.return_value = '0ab1ac0a-2867-402d'
>> > +        event.return_value = True
>> > +
>> > +        eventsObj = EventsSubClass()
>> > +        eventsObj._message_queue = kafka
>> > +        res = mock.MagicMock()
>> > +        res.body = {}
>> > +        res.status = 0
>> > +        eventsObj.on_post(self._generate_req(), res)
>> > +        self.assertEqual(falcon.HTTP_204, res.status)
> The on_post method is not setting the body in the method.
> 
> 
> Reply to this email directly or view it on GitHub
> <https://github.com/hpcloud-mon/monasca-events-api/pull/2/files#r32675515> .
>